VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitNavigationActivity.java

```diff
@@ -154,11 +154,11 @@ private void setCurrentUnit(CourseComponent component){
 
         environment.getDatabase().updateAccess(null, selectedUnit.getId(), true);
 
-        CourseComponent parent = component.getParent();
+        CourseComponent parentSectionComponent = component.getParent().getParent();
```

Comment:
We should really be storing as much of the full path as we can, but then only cutting it off when we go to actually do things with it like show UI. That way if we change how navigation works we know with as much detail as possible where the user used to be. Similarly, the web has a slightly different structure and always stores things at the subsection level, so that's what we get from there.